src/test.js

```diff
@@ -677,8 +709,10 @@ export function internalStop( test ) {
 
 // Forcefully release all processing holds.
 function internalRecover( test ) {
-	test.semaphore = 0;
-	internalStart( test );
+	if ( config.blocking ) {
```

Comment:
I'm not sure, but it seems like there may be some edge cases affected. At minimum, such a change doesn't seem necessary for this PR and should remain independent.